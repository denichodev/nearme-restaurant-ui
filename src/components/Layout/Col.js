import React from 'react';
import { number, node } from 'prop-types';

const classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
};

const getClassNames = props => {
  const extraClassNames = [];

  if (props.className) {
    extraClassNames.push(props.className);
  }

  return Object.keys(props)
    .filter(key => classMap[key])
    .map(key => `${classMap[key]}-${props[key]}`)
    .concat(extraClassNames)
    .join(' ');
};

const Col = props => {
  const cssClass = getClassNames(props);
  return <div className={cssClass}>{props.children}</div>;
};

Col.propTypes = {
  xs: number,
  sm: number,
  md: number,
  lg: number,
  xsOffset: number,
  smOffset: number,
  mdOffset: number,
  lgOffset: number,
  children: node.isRequired,
};

export default Col;
