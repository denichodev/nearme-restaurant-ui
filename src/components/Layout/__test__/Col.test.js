import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Col from '../Col';

describe('<Col />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <Col>
        <div>Children</div>
      </Col>,
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  test('has xs-* classnames according the prop', () => {
    const props = {
      xs: 3,
      xsOffset: 3,
    };

    const wrapper = shallow(
      <Col {...props}>
        <div>Children</div>
      </Col>,
    );

    expect(wrapper.find('.col-xs-3')).toHaveLength(1);
    expect(wrapper.find('.col-xs-offset-3')).toHaveLength(1);
  });

  test('has sm-* classnames according the prop', () => {
    const props = {
      sm: 3,
      smOffset: 3,
    };

    const wrapper = shallow(
      <Col {...props}>
        <div>Children</div>
      </Col>,
    );

    expect(wrapper.find('.col-sm-3')).toHaveLength(1);
    expect(wrapper.find('.col-sm-offset-3')).toHaveLength(1);
  });

  test('has md-* classnames according the prop', () => {
    const props = {
      md: 3,
      mdOffset: 3,
    };

    const wrapper = shallow(
      <Col {...props}>
        <div>Children</div>
      </Col>,
    );

    expect(wrapper.find('.col-md-3')).toHaveLength(1);
    expect(wrapper.find('.col-md-offset-3')).toHaveLength(1);
  });

  test('has lg-* classnames according the prop', () => {
    const props = {
      lg: 3,
      lgOffset: 3,
    };

    const wrapper = shallow(
      <Col {...props}>
        <div>Children</div>
      </Col>,
    );

    expect(wrapper.find('.col-lg-3')).toHaveLength(1);
    expect(wrapper.find('.col-lg-offset-3')).toHaveLength(1);
  });

  test('append new classname', () => {
    const props = {
      className: 'hello-world',
    };

    const wrapper = shallow(
      <Col {...props}>
        <div>Children</div>
      </Col>,
    );

    expect(wrapper.find('.hello-world')).toHaveLength(1);
  });
});
