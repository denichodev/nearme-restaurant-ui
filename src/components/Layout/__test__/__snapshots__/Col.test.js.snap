// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> renders correctly 1`] = `
<div
  className=""
>
  <div>
    Children
  </div>
</div>
`;
